tests/accelerators/test_ipu.py

```diff
@@ -561,7 +562,7 @@ def test_accelerator_ipu_with_devices():
 
     trainer = Trainer(accelerator="ipu", devices=8)
 
-    assert trainer.ipus == 8
+    assert trainer.num_devices == 8
     assert isinstance(trainer.strategy, IPUStrategy)
     assert isinstance(trainer.accelerator, IPUAccelerator)
```

Comment:
```suggestion
    assert isinstance(trainer.accelerator, IPUAccelerator)
    assert trainer.num_devices == 8
    assert isinstance(trainer.strategy, IPUStrategy)
```

Should the check for `trainer.accelerator` be before `trainer.num_devices == 8`?

Reply:
will it make any difference?

Reply:
@krshrimali I think they'll work the same, we are asserting both the types and device numbers. 

If logically if asserting accelerator type first makes more sense to you, I can change that.

Reply:
Sounds good to me, @DuYicong515 - thank you for the reasoning. Please don't change anything, it all looks good to me :)